src/app/portlets/content-types/fields/service/field-drag-drop.service.ts

```diff
@@ -216,19 +264,31 @@ export class FieldDragDropService {
     }
 
     private shouldAccepts(
-        _el: HTMLElement,
-        source: HTMLElement,
-        _handle: HTMLElement,
+        el: HTMLElement,
+        target: HTMLElement,
+        _source: HTMLElement,
         _sibling: HTMLElement
     ): boolean {
-        return source.dataset.dragType !== 'source';
+        const columnsCount = target.parentElement.querySelectorAll('.row-columns__item').length;
+        const isColumnField = el.dataset.clazz === COLUMN_BREAK_FIELD.clazz;
+        const cantAddColumn = isColumnField && columnsCount >= 4;
```

Comment:
maybe it should go into a method something like 'checkMaxColumnsByRow'

Reply:
I added the 4 into a const with a more explained name.